### ISSUE TITLE
DOTween Pro, path editor and local movement gizmos

### ISSUE THREAD

**Issue by Zamaroht** [Demigiant/dotween#87]
Hey, I recently got DOTween Pro. I'm noticing an issue with how the paths are displayed in the scene view when using the "Local Movement" option and moving/rotating the parent.

The behaviour is as expected and the element moves correctly through the path; however the path displayed is not rotating, and during play mode is not getting its position modified.

As a simple way to reproduce it:
- Open the DOTweenPath scene
- Create a new empty GameObject
- Move the Cube B inside it
- Tick "Local Movement" in cube B's DoTweenPath component
- Move and rotate this new parent gameObject

At this stage, if you click cube B again, you will see in the scene view that the path moved to the new correct position, however the rotation is not applied. If you press play, the path position gets reset to its original position aswell. The cube moves in the transformed position/rotation however, as its expected.

The versions shown in the utility panel are:
DoTween v1.1.135 [Release build]
DoTweenPro v0.9.470

Thanks!

**Comment by Zamaroht** [comment]
Actually, I'm finding some inconsistent behaviour when rotating a parent GameObject with a path inside, and then doing PathGetPoint() on the DOTweenPath, in some cases I get the transformed points but not in others, I could provide further examples if needed. I'm trying all this by using the visual editor.

**Comment by Demigiant** [comment]
Hi Nicolas,

Your mail arrived in the evening so I read it only now. It's 2AM so I'm
going to test it out and answer more thoroughly tomorrow.

Cheers,
Daniele

**Comment by Zamaroht** [comment]
Hey Daniele, thank you for the heads up. Could you manage to reproduce it? Any news?
Thank you!

**Comment by Demigiant** [comment]
Hey Nicolas,

Apologies for the late reply, these days of August are pretty complicated for me (I didn't even realize we were on github the first time :B).

I checked and could reproduce your issue, which for some reason is the first time someone mentions. Indeed the drawing of paths ignores the parent's rotation. This was actually by design, but I realize that it shouldn't be. I'll try to change this behaviour asap.

Cheers!

**Comment by Zamaroht** [comment]
No problem! Great to hear!
Looking forward to the update, thanks :)

**Comment by Zamaroht** [comment]
Hello! Any updates about this issue? Thanks!

**Comment by Demigiant** [comment]
Hi,

I'm sorry, I tried but it's really not working with the management of Unity editor's handles. I let it stay a few days to try another approach, which I tried Saturday, but that didn't work either. I have another idea, but I won't be able to do it until next Friday (because I need another half day at least). I'm afraid I forgot that this was indeed the reason why it doesn't rotate by design (except at runtime).
Can you work around it in the meantime?

**Comment by Zamaroht** [comment]
Hey Daniele, thank you for the heads up!

To give you a bit of context, we are planning on making a map composed of different prefabs; some of these prefabs contain children gameObjects with DOTweenPaths inside, that we use during gameplay.
We have yet not started working on the final map and don't have an estimated time, but I guess we will begin in the next few weeks.
The rotation is something we need since the level designers will be placing the prefabs themselves (and potentially rotating them), and this process should be as hassle free as possible. I guess we could work around it by making an editor script that saves the waypoints positions and then instantiate the path at runtime? Well, we will figure something out if the time comes :)

**Comment by Demigiant** [comment]
Ah, if they're just using prefabs with paths already there, there might be a trick (as long as they don't need to change the paths in the editor). Since the parent's rotation is taken into account at runtime (because the trouble is with the editor part), after the tween has been created, you could use a two steps process:
- during Awake, BEFORE the tween is created (so you'll need a Component whose ScriptExecutionOrder is upper than normal) you should reset the parent's rotation. That way the tween sets up with the default rotation
- during Start, you should set back to its previous one. That way, the path will be rotated correctly

**Comment by Zamaroht** [comment]
Yep, sounds like that should work wonders in our scenario! We are only making use of the editor when creating the prefabs themselves. I'll give it a shot and get back, thank you!

**Comment by Zamaroht** [comment]
Well, I'm afraid it's not working :( I just sent you a link with a minimal example through your website's contact form.

**Comment by kevinkassimo** [comment]
Having a similar trouble, after rotating the parent object, the moving path showed up somewhere else...

**Comment by atadogan** [comment]
I am having the same issue wondering if theres a way to fix this? 

For me position is working but rotation isn't following parent.
